src/v/cluster/read_replica_manager.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2020 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cloud_storage/remote.h"
+#include "s3/client.h"
+
+#include <seastar/core/abort_source.hh>
+
+namespace cluster {
+
+class read_replica_manager {
+public:
+    read_replica_manager(cloud_storage::remote&);
+
+    ss::future<errc> set_remote_properties(
```

Comment:
nit: add doc comment, also the `cfg` is out parameter, and it's not quite clear that it'd be updated. Maybe you can make it the first parameter or change name to `set_remote_properties_in_config` or smthng?

Reply:
Or you can pass bucket and as in c-tor.

Reply:
Changed the method name to `set_remote_properties_in_config`, made `cfg` the first param and added a comment.